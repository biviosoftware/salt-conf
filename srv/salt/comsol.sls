comsol-pkgs:
  pkg.installed:
    - pkgs:
      - libXtst
      - redhat-lsb-core
      - webkitgtk
      - xorg-x11-xauth
      - xterm
