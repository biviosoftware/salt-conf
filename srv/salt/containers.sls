base_pkgs:
  bivio.pkg_installed:
    pkgs:
      - docker
      - emacs-nox
      - lsof
      - screen
      - tar
      - telnet

postgresql_container:
  bivio.docker_container:
    - container_name: postgresql
    - image_name: radiasoft/postgres
    - volumes:
      - /var/lib/postgresql/data
      - [ /var/lib/postgresql/run, /run/postgresql ]
    - user: postgres
    - ports:
      - [ 5432 5432 ]
    - init:
      env:
        - [ POSTGRES_PASSWORD, {{ postgres.admin_pass }} ]
        - [ JPY_PSQL_PASSWORD, {{ jupyterhub.db_pass }} ]
      cmd: /radia-init.sh
      sentinel: /var/lib/postgresql/data/PG_VERSION

jupyter_singleuser:
  bivio.docker_image:
    - name: radiasoft/jupyter-singleuser

jupyterhub_config:
  bivio.plain_file:
    - name: /var/lib/jupyterhub/conf/jupyterhub_config.py
    - user: {{ pillar.docker_container_user }}

jupyterhub:
  bivio.docker_container:
    - image: radiasoft/jupyterhub
    - links:
        - postgresql
    - want_docker_sock: True
    - volumes:
        - [ /var/lib/jupyterhub/conf, /srv/jupyterhub/conf ]
    - user: root
    - ports:
        - [ 5692, 8000 ]
    - cmd: jupyterhub -f /srv/jupyterhub/conf/jupyter_config.py
    - after:
        - postgresql
    - require:
        - bivio.postgresql
        - bivio.jupyter_singleuser
        - bivio.jupyterhub_config
