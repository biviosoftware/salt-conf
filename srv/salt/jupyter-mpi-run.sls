jupyter_cluster_start:
  radia.cluster_start: []
