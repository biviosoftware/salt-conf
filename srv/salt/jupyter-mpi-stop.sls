jupyter_cluster_stop:
  radia.cluster_stop: []
