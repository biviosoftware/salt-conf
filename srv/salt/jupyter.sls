jupyter_image:
  radia.docker_image:
    - image_name: '{{ pillar.jupyter.image_name }}'

jupyter_nfs:
  radia.nfs_mount:
    - local_dir: '{{ pillar.jupyter.nfs_local_d }}'
    - remote_dir: '{{ pillar.jupyter.nfs_remote_d }}'
    - user: '{{ pillar.jupyter.host_user }}'
    - group: '{{ pillar.jupyter.guest_user }}'
    - dir_mode: 700
