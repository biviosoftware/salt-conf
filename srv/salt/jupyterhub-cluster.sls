jupyterhub_cluster_start:
  radia.cluster_start: []
