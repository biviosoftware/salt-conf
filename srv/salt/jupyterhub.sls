#
# TODO(robnagler) configure backups for postgres
#
postgresql_jupyterhub_container:
  radia.docker_container:
    - container_name: '{{ pillar.jupyterhub.postgresql_name }}'
    - cmd: postgres
    - guest_user: postgres
    - image_name: '{{ pillar.jupyterhub.postgresql_image }}'
    - init:
        env:
          - [ POSTGRES_PASSWORD, {{ pillar.postgresql_jupyterhub.admin_pass }} ]
          - [ JPY_PSQL_PASSWORD, {{ pillar.jupyterhub.db_pass }} ]
        cmd: bash /radia-init.sh
        sentinel: {{ pillar.jupyterhub.postgresql_host_data_d }}/PG_VERSION
    - ports:
      - [ 5432, 5432 ]
    - volumes:
      - [ '{{ pillar.jupyterhub.postgresql_host_data_d }}', /var/lib/postgresql/data ]
      - [ '{{ pillar.jupyterhub.postgresql_host_run_d }}', /run/postgresql ]

jupyterhub_config:
  radia.plain_file:
    - file_name: '{{ pillar.jupyterhub.host_conf_f }}'
    - source: salt://jupyterhub/jupyterhub_config.py
    - user: root
    - group: root
    - zz:
        jupyter_image: '{{ pillar.jupyter.image_name }}'

jupyterhub_container:
  radia.docker_container:
    - after:
        - {{ pillar.jupyterhub.postgresql_name }}
    - container_name: jupyterhub
    - cmd: jupyterhub -f '{{ pillar.jupyterhub.guest_conf_f }}'
    - image_name: '{{ pillar.jupyterhub.image_name }}'
    - guest_user: {{ pillar.jupyterhub.guest_user }}
    - host_user: {{ pillar.jupyterhub.host_user }}
    - links:
        - {{ pillar.jupyterhub.postgresql_name }}
    - ports:
        - [ {{ pillar.jupyterhub.host_port }}, {{ pillar.jupyterhub.guest_port }} ]
    - volumes:
        - [ {{ pillar.jupyterhub.host_conf_d }}, {{ pillar.jupyterhub.guest_conf_d }} ]
        - [ {{ pillar.jupyter.nfs_local_d }}, {{ pillar.jupyter.nfs_local_d }} ]
    - want_docker_sock: True
    - watch:
        - jupyterhub_config
