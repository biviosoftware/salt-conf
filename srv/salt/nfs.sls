/var/nfs/apa11/home:
  pkg.installed:
    - pkgs:
      - nfs-utils
  file.directory:
    - user: root
    - group: root
    - mode: 755
    - makedirs: True
  mount.mounted:
    - device: apa11b.bivio.biz:/home
    - fstype: nfs4
    # Work around: https://github.com/saltstack/salt/issues/18630
    - opts: rw,relatime,vers=4.0,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,port=0,timeo=600,retrans=2,sec=sys,clientaddr={{ grains.fqdn_ip4[0] }},local_lock=none,addr=192.168.1.11
