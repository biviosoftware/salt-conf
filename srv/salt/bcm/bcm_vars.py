#!/usr/bin/env python
from __future__ import print_function


def main(argv):
    assert len(argv) == 1, \
        'usage: bcm_vars pillars|grains|<file>.yml'
    prefix = argv[0]
    if prefix in ('grains', 'pillar'):
        a = _value_salt(prefix)
    else:
        a = _value_yaml(prefix)
    _print_value(*a)
    return 0


def _print_array(name, value, atype):
    print('declare -{} {}=('.format('A' if issubclass(atype, dict) else 'a', name))
    for k, v in value.items():
        print("[{}]='{}'".format(k, v))
    print(');')


def _print_value(prefix, value):
    assert isinstance(value, (dict, list)), \
        '{}: unable to handle type {}'.format(value, type(value))
    as_array = {}
    gen = value.items() if isinstance(value, dict) else zip(xrange(len(value)), value)
    for k, v in gen:
        pk = prefix + '_' + str(k)
        if isinstance(v, (dict, list)):
            _print_value(pk, v)
            # Only can print as_array if all values are scalar
            as_array = None
        else:
            v = _scalar(v)
            if as_array is not None:
                as_array[k] = v
            print("{}='{}';".format(pk, v))
    if as_array is not None:
        _print_array(prefix, as_array, type(value))


def _scalar(v):
    if isinstance(v, bool):
        return '1' if v else ''
    elif v is None:
        return ''
    # Might be a float or int so convert now
    return str(v)


def _value_salt(prefix):
    import salt.client
    c = salt.client.Caller()
    return prefix, c.function(prefix + '.items')


def _value_yaml(filename):
    import re
    m = re.search(r'(\w+)\.yml$', filename)
    assert m, \
        '{}: not YAML or pillars or grains'.format(filename)
    import yaml
    return m.group(1), yaml.load(open(filename))


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv[1:]))
