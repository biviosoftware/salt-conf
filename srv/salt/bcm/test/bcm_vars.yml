a:
  x: 1
  y: 2
b:
  x: [1, 2, 3]
  y:
    -
      p: 1
      q: 2
    -
      r:
