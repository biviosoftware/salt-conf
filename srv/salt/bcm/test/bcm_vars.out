bcm_vars_a_y='2';
bcm_vars_a_x='1';
declare -A bcm_vars_a=(
[y]='2'
[x]='1'
);
bcm_vars_b_y_0_q='2';
bcm_vars_b_y_0_p='1'"'"'';
declare -A bcm_vars_b_y_0=(
[q]='2'
[p]='1'"'"''
);
bcm_vars_b_y_1_r=;
declare -A bcm_vars_b_y_1=(
[r]=
);
bcm_vars_b_x_0='1';
bcm_vars_b_x_1='2';
bcm_vars_b_x_2='3';
declare -a bcm_vars_b_x=(
[0]='1'
[1]='2'
[2]='3'
);
